packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -207,10 +262,33 @@ export class TaskDefinition extends TaskDefinitionBase {
    * The task will have a compatibility of EC2+Fargate.
    */
   public static fromTaskDefinitionArn(scope: Construct, id: string, taskDefinitionArn: string): ITaskDefinition {
+    return TaskDefinition.fromTaskDefinitionAttributes(scope, id, { taskDefinitionArn: taskDefinitionArn });
+  }
+
+  /**
+   * Create a task definition from a task definition reference
+   */
+  public static fromTaskDefinitionAttributes(scope: Construct, id: string, attrs: TaskDefinitionAttributes): ITaskDefinition {
     class Import extends TaskDefinitionBase {
-      public readonly taskDefinitionArn = taskDefinitionArn;
-      public readonly compatibility = Compatibility.EC2_AND_FARGATE;
-      public readonly executionRole?: iam.IRole = undefined;
+      public readonly taskDefinitionArn = attrs.taskDefinitionArn;
+      public readonly compatibility = attrs.compatibility ?? Compatibility.EC2_AND_FARGATE;
+      public readonly executionRole = attrs.executionRole;
+
+      public get networkMode(): NetworkMode {
+        if (attrs.networkMode == undefined) {
+          throw new Error('NetworkMode is available only if it is given when importing the TaskDefinition.');
```

Comment:
Almost there. This error message makes sense to a user if they are calling `taskDefinition.networkMode` directly.

However, in 99% of cases the user will be PASSING this TaskDefinition to some other construct or function, and THAT one will be calling `.networkMode`. So think of it like this:

```ts
// User writes this:
const taskDefintiion = TaskDefintion.fromTaskDefinitionAttributes(...);
startNewService(taskDefinition);

// User gets error:
"NetworkMode is available only if it is given..."
```

And the user's thought will be "where is this NetworkMode coming from?"

A more accurate and useful message would be something like:

```
This operation requires the TaskDefinition's network mode to be known. Add the 'networkMode' parameter to the call to 'TaskDefinition.fromNetworkAttributes(...)'
```

Or somethign.

Reply:
Got it. Will update.